ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -11564,8 +11564,11 @@ private Operator genTablePlan(String alias, QB qb) throws SemanticException {
           }
         }
         // Obtain inspector for schema
-        StructObjectInspector rowObjectInspector = (StructObjectInspector) tab
-            .getDeserializer().getObjectInspector();
+        final Deserializer deserializer = tab.getDeserializer();
+        StructObjectInspector rowObjectInspector = (StructObjectInspector) deserializer.getObjectInspector();
+
+        // Specific Handling for Iceberg serde, which sets specific properties in SessionConf.
```

Comment:
No need to mention Iceberg here, the interface documentation should be clear instead what this method is doing.